eng/common/pipelines/templates/steps/verify-links.yml

```diff
@@ -1,19 +1,25 @@
 parameters:
   Directory: 'not-specified'
-  IgnoreLinksFile: "$(Build.SourcesDirectory)/eng/ignore-links.txt"
-  
+  IgnoreLinksFile: '$(Build.SourcesDirectory)/eng/ignore-links.txt'
+  WorkingDirectory: '$(System.DefaultWorkingDirectory)'
+  ScriptDirectory: 'eng/common/scripts'
+  Recursive: $false
+  CheckLinkGuidance: $true
+  Urls: 'Get-ChildItem -Path ./ -Recurse -Include *.md'
+
 steps:
-- task: PowerShell@2
-  displayName: Link verification check
-  inputs:
-    pwsh: true
-    workingDirectory: $(Build.SourcesDirectory)/${{ parameters.Directory }}
-    filePath: eng/common/scripts/Verify-Links.ps1 
-    arguments: >
-      -urls $(dir -r -i *.md) 
-      -rootUrl "file://$(Build.SourcesDirectory)/${{ parameters.Directory }}" 
-      -recursive: $false 
-      -ignoreLinksFile ${{ parameters.IgnoreLinksFile }}
-      -branchReplaceRegex "^($env:SYSTEM_PULLREQUEST_SOURCEREPOSITORYURI.*/(?:blob|tree)/)master(/.*)$"
-      -branchReplacementName $env:SYSTEM_PULLREQUEST_SOURCECOMMITID
-      -devOpsLogging: $true
+  - task: PowerShell@2
+    displayName: Link verification check
+    inputs:
+      pwsh: true
+      workingDirectory: '${{ parameters.WorkingDirectory }}/${{ parameters.Directory }}'
+      filePath: ${{ parameters.ScriptDirectory }}/Verify-Links.ps1
+      arguments: >
+        -urls ${{ parameters.Urls }}
+        -rootUrl "file://${{ parameters.WorkingDirectory }}/${{ parameters.Directory }}"
+        -recursive: ${{ parameters.Recursive }} 
+        -ignoreLinksFile ${{ parameters.IgnoreLinksFile }}
+        -branchReplaceRegex '^$env:SYSTEM_PULLREQUEST_SOURCEREPOSITORYURI.*/(?:blob|tree)/)master(/.*)$'
```

Comment:
I don't think single quotes here is going to work as we chatted in IM. This should be double quotes and I'd still like to make these template parameters, although I'm fine if you are just doing this temporarily for testing.

Reply:
For env vars, they have to be double quotes, yes, to support variable replaceent. But you have to write it like so when using a scope:

```powershell
"${env:SYSTEM_PULLREQUEST}.*/ ..."
```

Always use single quotes when you do *not* want to support variable replacements. Use double quotes for everything else. It's basically a security precaution, but can also just help avoid surprises where you didn't intend for variable replacement to occur.

Reply:
Right, I am in the middle of testing. So it might look weird which I mix the use sometimes. More to learn and remember here.
Thanks for the knowledge share.